graylog2-web-interface/src/components/widgets/Widget.jsx

```diff
@@ -204,37 +207,34 @@ const Widget = React.createClass({
       return <span />;
     }
     const showConfigModal = (
-      <WidgetConfigModal ref="configModal"
-                         dashboardId={this.props.dashboardId}
+      <WidgetConfigModal dashboardId={this.props.dashboardId}
                          widget={this.props.widget}
                          boundToStream={this._isBoundToStream()} />
     );
 
     const editConfigModal = (
-      <WidgetEditConfigModal ref="editModal"
-                             widget={this.props.widget}
+      <WidgetEditConfigModal widget={this.props.widget}
                              onUpdate={this.updateWidget} />
     );
 
-    let disabledTooltip = null;
-    if (this.props.streamIds != null && this.props.widget.config.stream_id &&
-        !this.props.streamIds[this.props.widget.config.stream_id]) {
-      disabledTooltip = 'The stream is not available, cannot replay search.';
-    }
+    const disabledReplay = !((!this.props.widget.config.stream_id &&
+      this.isPermitted(this.state.currentUser.permissions,
+        ['searches:absolute', 'searches:keyword', 'searches:relative'])) ||
+      (this.props.widget.config.stream_id && this.props.streamIds != null &&
+      this.props.streamIds[this.props.widget.config.stream_id]));
```

Comment:
I personally find this sentence hard to read. Could we maybe extract it into a couple of variables or functions? In that way we can split the condition and be able to understand it easier.